### ISSUE TITLE
Suboptimal codegen for Vector3 -> Vector128 conversion`

### ISSUE THREAD

**Issue by EgorBo** [dotnet/runtime#92671]
```cs
Vector128<float> Test(Vector3 v) => v.AsVector128();
```
Emits:
```asm
; Method Runtime_90508:Test(System.Numerics.Vector3):System.Runtime.Intrinsics.Vector128`1[float]:this (FullOpts)
       vzeroupper 
       vmovsd   xmm0, qword ptr [r8]
       vinsertps xmm0, xmm0, dword ptr [r8+0x08], 40
       vinsertps xmm0, xmm0, xmm0, 56
       vmovups  xmmword ptr [rdx], xmm0
       mov      rax, rdx
       ret      
; Total bytes of code: 29
```
Looks like we don't need double insertps here, we can load the Vector3 once using a 16-byte load (at least on x64 abis) and clear its 4th element.

**Comment by tannergooding** [comment]
> Looks like we don't need double insertps here

Right. It actually should've been folded. I'm unsure why it wasn't.

We should be emitting just the following, because `40` is `0b0010_1000` (insert to index 2, zero index 3):
```
vmovsd   xmm0, qword ptr [r8]
vinsertps xmm0, xmm0, dword ptr [r8+0x08], 40
```

The `vinsertps xmm0, xmm0, xmm0, 56` is superfluous and is just rezeroing index 3 (`0b0011_1000` is insert to index 3, zero index 3).

> we can load the Vector3 once using a 16-byte load (at least on x64 ABIs) and clear its 4th element.

A single 16-byte load is tricky since that requires assumptions around stack layout and other factors. There is nothing guaranteeing that the operand is from a stack slot and there are ways (such as function pointers) that can bypass the normal behavior.

I think keeping it as a two part memory load is then safer and the same general instruction sequence.

**Comment by tannergooding** [comment]
> I'm unsure why it wasn't.

I'd guess that there is a missing check for if the last insert was also zeroing that element or something trivial that's blocking it.

**Comment by tfenise** [comment]
Similarly,
```C#
static Vector128<float> M(ref Vector2 x) => x.AsVector128();
```
```asm
vmovsd   xmm0, qword ptr [rsi]
vinsertps xmm0, xmm0, xmm0, 60
vmovups  xmmword ptr [rdi], xmm0
mov      rax, rdi
ret
```
The `vinsertps` is unnecessary.
